### ISSUE TITLE
Make a new release

### ISSUE THREAD

**Issue by salvatorebenedetto** [imgui-rs/imgui-rs#807]
Given the recent upgrade of the imgui library, can we get a new imgui-rs release? 

CC: @sanbox-irl

**Comment by sanbox-irl** [comment]
Before a new release, I'd prefer to bind the new multi-select tools and (perhaps) better understand what, if anything, we need to do for the error correction stuff. If a hero binds the new multi-select stuff (https://github.com/ocornut/imgui/wiki/Multi-Select) that would be sufficient for me to make a release.

I probably won't have time to do it myself till around Thanksgiving since I put in ~20 hours recently. I'm currently hoping for someone to help us out with it.

For now, I would use a git dependency to use the current imgui-rs here!

**Comment by petar-andrejic** [comment]
Hi, I noticed there hasn't been a new release in two years, but there is still some activity in the repo.

Is the current release still fine for purposes that don't involve the new APIs added in the meantime? I.e. it hasn't gotten out of sync with the backend graphics libraries?

Cheers.

**Comment by sanbox-irl** [comment]
Yeah, unfortunately I've not had the time to do active maintenance. ImGui is a hard library to maintain because:

1. The underlying C++ library constantly updates (impressive and good!).
2. The linked graphics libraries must update consistently. This means, in my opinion, the best course for users is actually to vendor all the glue code between ImGui and their renderer backend of choice, as they'd have to with a custom written Renderer, that way they don't need to wait for minor releases which merely bump dependencies.
3. Finally, with the library currently in a good state API wise with Rust's current features, all of the above work is frankly drudgery, which burns through maintainers, especially once they realize it's endless.

Ultimately, I suspect a better version of this library would be entirely automatically generated from the C++, with some better support for wgpu/glow/etc bindings being updates for trivial dependency management. I do think our API is somewhat overwrought compared to the C++, but we deeply suffer from the lack of optional arguments in Rust. I suspect when <https://github.com/rust-lang/rust/issues/132162> lands on stable, we can take another crack at making the API better. That being said, this will just lessen issues 1-3, which will continue to be the problem with this library.
